api/tests/Api/SnapshotTests/EndpointPerformanceTest.php

```diff
@@ -15,43 +16,52 @@
 use function PHPUnit\Framework\equalTo;
 use function PHPUnit\Framework\greaterThanOrEqual;
 use function PHPUnit\Framework\isEmpty;
+use function PHPUnit\Framework\lessThan;
 use function PHPUnit\Framework\lessThanOrEqual;
 use function PHPUnit\Framework\logicalAnd;
 
+const MAX_EXECUTION_TIME_SECONDS = 0.5;
+
 /**
  * @internal
  */
-class EndpointQueryCountTest extends ECampApiTestCase {
+class EndpointPerformanceTest extends ECampApiTestCase {
     /**
      * @throws ClientExceptionInterface
      * @throws DecodingExceptionInterface
      * @throws RedirectionExceptionInterface
      * @throws ServerExceptionInterface
      * @throws TransportExceptionInterface
      */
-    public function testNumberOfQueriesDidNotChangeForStableEndpoints() {
+    public function testPerformanceDidNotChangeForStableEndpoints() {
         $numberOfQueries = [];
         $responseCodes = [];
+        $queryExecutionTime = [];
         $collectionEndpoints = self::getCollectionEndpoints();
         foreach ($collectionEndpoints as $collectionEndpoint) {
             if ('/users' !== $collectionEndpoint && !str_contains($collectionEndpoint, '/content_node')) {
-                list($statusCode, $queryCount) = $this->measurePerformanceFor($collectionEndpoint);
+                list($statusCode, $queryCount, $executionTimeSeconds) = $this->measurePerformanceFor($collectionEndpoint);
                 $responseCodes[$collectionEndpoint] = $statusCode;
                 $numberOfQueries[$collectionEndpoint] = $queryCount;
+                $queryExecutionTime[$collectionEndpoint] = $executionTimeSeconds;
             }
 
             if (!str_contains($collectionEndpoint, '/content_node')) {
                 $fixtureFor = $this->getFixtureFor($collectionEndpoint);
-                list($statusCode, $queryCount) = $this->measurePerformanceFor("{$collectionEndpoint}/{$fixtureFor->getId()}");
+                list($statusCode, $queryCount, $executionTimeSeconds) = $this->measurePerformanceFor("{$collectionEndpoint}/{$fixtureFor->getId()}");
                 $responseCodes["{$collectionEndpoint}/item"] = $statusCode;
                 $numberOfQueries["{$collectionEndpoint}/item"] = $queryCount;
+                $queryExecutionTime["{$collectionEndpoint}/item"] = $executionTimeSeconds;
             }
         }
 
-        $not200Responses = array_filter($responseCodes, fn ($value) => 200 != $value);
-        assertThat($not200Responses, isEmpty());
+        $tooLongQuery = array_filter($responseCodes, fn ($value) => 200 != $value);
+        assertThat($tooLongQuery, isEmpty());
+
+        $endpointsWithTooLongExecutionTime = array_filter($queryExecutionTime, fn ($value) => MAX_EXECUTION_TIME_SECONDS < $value);
+        assertThat($endpointsWithTooLongExecutionTime, isEmpty());
```

Comment:
A bit difficult to debug, as the output is simply "Failed asserting that an array is empty." if one or multiple queries are running too long, without any further information which queries failed.
Otherwise seems do work as intended.

Reply:
fixed by comparing to []